azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -283,6 +287,31 @@ public void setExecutorActive(final boolean isActive, final String host, final i
           "Set active action ignored. Executor is already " + (isActive ? "active" : "inactive"));
     }
     this.active = isActive;
+    if (!this.active) {
+      // When deactivating this executor, this call will return until every thread in {@link
```

Comment:
Change "will return until every thread in {@link#createFlowRunner} to finish" to "will wait to return until every thread in {@link#createFlowRunner}  has finished".